### ISSUE TITLE
[BUG] Word not commit was deleted on QT app when mouse move out

### ISSUE THREAD

**Issue by zenfas** [ibus/ibus-m17n#55]
[BUG] Word not commit was deleted (instead of commit and keep it like GTK app) on QT app when mouse move out

**To Reproduce**
1. Type Vietnamese  on Kate or FeatherPad (QT app)
2. Mouse move out and click on another app

**Expected behavior**
Current preedit word on Kate should be committed 

**Screenshots or videos**
![Peek 2022-12-15 16-08 Xubuntu](https://user-images.githubusercontent.com/66258788/207821553-c95646d3-1e73-4186-9aa5-0fb60eee761c.gif)

**ibus-m17n version?** 1.4.18

**m17n-db versions?** 1.8.0-3

**ibus version?* 1.5.27-4

**Distribution and version?** Xubuntu 23.04

**Desktop and version?** XFCE

**Xorg or Wayland?** X

**Comment by mike-fabian** [comment]
I guess that is not a bug in ibus-m17n because the same behaviour happens with other ibus input methods which use IBus.PreeditFocusMode.COMMIT. Usually the preedit should be committed when that option is used and it seems to work in most applications but not in kate.

**Comment by mike-fabian** [comment]
By the way, which input method exactly of the ibus-m17n input methods are you using?

**Comment by zenfas** [comment]
I use VN Telex. This issue also happen in ibus-unikey
Not only Kate, FeatherPad also
Fcitx has same issue fcitx/fcitx5/issues#687

**Comment by mike-fabian** [comment]
Yes, I think this is an issue of either Qt5 or ibus, not of these specific ibus input methods. Because it happens for many ibus input  methods. 

And it does not happen when the focus is moved to another Qt5 application instead of to a Gtk application.

The last comments in  https://github.com/fcitx/fcitx5/issues/687 also seem to indicate that this is not a problem of the specific input method. Most likely this is an issue in ibus and/or the Qt5 ibus input module.

**Comment by mike-fabian** [comment]
https://user-images.githubusercontent.com/2330175/208103541-788334bb-73f7-438f-94bf-8a3a934ba448.mp4

**Comment by mike-fabian** [comment]
And this video shows that it does not happen when the focus is moved between anki and kate (both anki and kate are Qt5 applications):

https://user-images.githubusercontent.com/2330175/208104175-80e5cb9b-33d1-409f-b29b-dba9e594243e.mp4

**Comment by zenfas** [comment]
@mike-fabian can you move this issue to ibus issue? Or I need create new one?
